MoviesforSale/app/src/main/java/com/ezequielc/moviesforsale/MainActivity.java

```diff
@@ -0,0 +1,77 @@
+package com.ezequielc.moviesforsale;
+
+import android.app.SearchManager;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.Fragment;
+import android.support.v4.view.MenuItemCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.SearchView;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+
+import com.ezequielc.moviesforsale.setup.DBAssetHelper;
+
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity
+        implements MovieListFragment.OnMovieSelectedListener{
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Database Setup
+        DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
+        dbSetup.getReadableDatabase();
+
+        // Adding MovieListFragment to the MainActivity
+        Fragment movieListFragment = MovieListFragment.newInstance(null, MainActivity.this);
+        getSupportFragmentManager()
+                .beginTransaction()
+                .add(R.id.fragment_container, movieListFragment, "list")
+                .commit();
+
+        handleIntent(getIntent());
+    }
+
+    // Once a Movie is clicked on Replaces Fragment with MovieDetailFragment
+    @Override
+    public void onMovieSelected(int id) {
+        Bundle bundle =  new Bundle();
+        bundle.putInt("selected_movie", id);
+        Fragment movieDetailFragment = MovieDetailFragment.newInstance(bundle);
+
+        getSupportFragmentManager()
+                .beginTransaction()
+                .addToBackStack(null)
+                .replace(R.id.fragment_container, movieDetailFragment)
```

Comment:
Great use of an interface!!!

Also, good idea to call `addToBackStack()` so the user can navigate back to the Movies list. If you wanted to, you could actually build your entire app with just one activity as long as you structure your fragment transactions carefully :)